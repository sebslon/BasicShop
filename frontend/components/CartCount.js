import styled from "styled-components";
import { CSSTransition, TransitionGroup } from "react-transition-group";

export default function CartCount({ count }) {
  return (
    <AnimationStyles>
      <TransitionGroup>
        <CSSTransition
          unmountOnExit
          className="count"
          classNames="count"
          key={count}
          timeout={{ enter: 300, exit: 300 }}
        >
          <Dot>{count}</Dot>
        </CSSTransition>
      </TransitionGroup>
    </AnimationStyles>
  );
}

const Dot = styled.div`
  background: var(--secondary);
  color: white;
  border-radius: 50%;
  padding: 0.5rem;
  line-height: 2rem;
  min-width: 3rem;
  margin-left: 1rem;

  font-feature-settings: "tnum";
  font-variant-numeric: tabular-nums;
`;

const AnimationStyles = styled.span`
  position: relative;

  .count {
    display: block;
    position: relative;
    transition: transform 0.3s;
    backface-visibility: hidden;
  }

  .count-enter {
    transform: scale(4) rotateX(0.5turn);
  }

  .count-enter-active {
    transform: rotateX(0);
  }

  .count-exit {
    top: 0;
    position: absolute;
    transform: rotateX(0);
  }

  .count-exit-active {
    transform: scale(4) rotateX(0.5turn);
  }
`;
