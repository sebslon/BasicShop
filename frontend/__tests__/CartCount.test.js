import { render, screen } from "@testing-library/react";
import wait from "waait";

import CartCount from "../components/CartCount";

describe("<CartCount />", () => {
  it("Renders", () => {
    render(<CartCount count={10} />);
  });

  it("Matches snapshot", () => {
    const { container } = render(<CartCount count={10} />);

    expect(container).toMatchSnapshot();
  });

  it("Updates via props", async () => {
    const { container, rerender } = render(<CartCount count={10} />);

    expect(container.textContent).toBe("10");

    rerender(<CartCount count={11} />);

    await wait(400);

    expect(container.textContent).toBe("11");
    expect(container).toMatchSnapshot();
  });
});
