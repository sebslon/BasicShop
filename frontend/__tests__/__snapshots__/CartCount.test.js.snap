// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount /> Matches snapshot 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-sc-1thwzyj-1 XjpSt"
  >
    <div>
      <div
        class="CartCount__Dot-sc-1thwzyj-0 bTkECI count"
      >
        10
      </div>
    </div>
  </span>
</div>
`;

exports[`<CartCount /> Updates via props 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-sc-1thwzyj-1 XjpSt"
  >
    <div>
      <div
        class="CartCount__Dot-sc-1thwzyj-0 bTkECI count count-enter-done"
      >
        11
      </div>
    </div>
  </span>
</div>
`;
